src/processor/Package.ts

```diff
@@ -83,4 +86,35 @@ export class Package {
       pullAt(sd.rules, rulesToRemove);
     });
   }
+
+  // Choice elements have a standard set of slicing rules applied to them by SUSHI.
+  // Therefore, it is not necessary to define that slicing using FSH when one of the choices exists.
+  private suppressChoiceSlicingRules(): void {
+    [...this.profiles, ...this.extensions].forEach(sd => {
+      const rulesToRemove: number[] = [];
+      sd.rules.forEach((rule, i, allRules) => {
+        if (rule instanceof ExportableCaretValueRule && rule.path.endsWith('[x]')) {
+          const pathStart = rule.path.replace(/\[x\]$/, '');
+          // check the four relevant caret paths and their default values, and
+          // check if one of the choices exists
+          if (
+            ((rule.caretPath === 'slicing.discriminator[0].type' &&
+              rule.value instanceof fshtypes.FshCode &&
+              rule.value.code === 'type') ||
+              (rule.caretPath === 'slicing.discriminator[0].path' && rule.value === '$this') ||
+              (rule.caretPath === 'slicing.ordered' && rule.value === false) ||
+              (rule.caretPath === 'slicing.rules' &&
+                rule.value instanceof fshtypes.FshCode &&
+                rule.value.code === 'open')) &&
+            allRules.some(
+              otherRule => otherRule.path != rule.path && otherRule.path.startsWith(pathStart)
+            )
+          ) {
+            rulesToRemove.push(i);
+          }
+        }
+      });
+      pullAt(sd.rules, rulesToRemove);
```

Comment:
This considers each standard choice slicing rule individually; _not_ as a set.  So if we had something like:
```
* value[x] ^slicing.discriminator[0].type = #value
* value[x] ^slicing.discriminator[0].path = $this
* value[x] ^slicing.ordered = false
* value[x] ^slicing.rules = #open
```
then this would _keep_ the first rule but remove the others.  I don't think it makes sense to output _partial_ choice slicing rules.  I think we need to consider them as a set.  If all 4 rules match, we remove them; otherwise, we keep them all.

Reply:
I'm a little shaky on this. As far as I can tell, if `value[x]` did happen to have just "slicing.ordered" defined on it and set to `true`, SUSHI would maintain that when automatically adding the type slicing. So if the user did:
```
Profile: MyObservation
Parent: Observation
* value[x] ^slicing.rules = #closed
* valueString = "hello"
```
The resulting slicing is:
```
"slicing": {
          "discriminator": [
            {
              "type": "type",
              "path": "$this"
            }
          ],
          "ordered": false,
          "rules": "closed"
        }
```
But then GoFSH is going to read that in, and not all rules will match, so it will output all four of the `CaretRules`, when really only the one that differs from the defaults, setting `rules` to `closed` is necessary.

Reply:
Ultimately it's a stylistic thing, I guess.  But I feel like having GoFSH output partial slicing definitions, based on inference of what SUSHI would do, could confuse users.  They might wonder, why did SUSHI only output the slicing ordered rule but not the others?  And I *feel* like if FSH Authors wanted to override parts of a choice slicing, they would likely define the whole thing -- partly because most authors probably don't even know how SUSHI handles this -- so they wouldn't know to just put in slicing.ordered and not worry about the rest.

I guess I'm just saying that I don't think partially overriding inferred slicings is a best practice or a reasonable thing to expect authors to do. Outputting the full slicing construct is not harmful and allows the full context of the slicing to be understood.  At least that's my thoughts on this. ;-)

Reply:
Yeah I'm fine with outputting more rules for clarity.